### PR TITLE
Add Decipher Trajectory methods

### PR DESCRIPTION
**Pull request by justjhong**
Implements the methods that require a pre-computed trajectory onto the Decipher model class.

cc @ANazaret

### CODE REVIEW

File: src/scvi/external/decipher/_model.py [hunk 1]
```diff
@@ -1,11 +1,11 @@
-import logging
 from collections.abc import Sequence
 
 import numpy as np
 import pyro
 import torch
 import torch.nn.functional as F
 from anndata import AnnData
+from sklearn.neighbors import KNeighborsRegressor
```
Comment (canergen) [inline src/scvi/external/decipher/_model.py:8]: We don't have a dependency of sklearn? Please use lazy import.

File: src/scvi/external/decipher/utils/__init__.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
-from ._utils import rotate_decipher_components
+from ._rotate import rotate_decipher_components
```
Comment (canergen) [inline src/scvi/external/decipher/utils/__init__.py:1]: Im confused about the rotate file? It doesn't exist right?

**Comment by canergen** [comment]
LGTM. See two point above, fix pre-commit missing empty line and add CHANGELOG.

File: src/scvi/external/decipher/utils/__init__.py (hunk 2 above)
Comment (justjhong) [reply src/scvi/external/decipher/utils/__init__.py:1]: it exists within the utils folder

**Pull request merged**